util/progress/printer.go

```diff
@@ -184,3 +186,36 @@ func WithOnClose(onclose func()) PrinterOpt {
 		opt.onclose = onclose
 	}
 }
+
+func dedupWarnings(inp []client.VertexWarning) []client.VertexWarning {
+	m := make(map[string]client.VertexWarning)
+	for _, w := range inp {
+		h := sha256.New()
```

Comment:
Should we use `hashstructure` in here? We don't need crypto hash and this way it doesn't break when new fields hare added.

Reply:
Good call out, I made the switch to `hashstructure`.

Reply:
Doesn't seem to work https://github.com/docker/buildx/actions/runs/9668610302/job/26673242251?pr=2551#step:7:822

Will take a closer look

Reply:
Maybe the `Source.Definition` would need to be cleared?